syntax = "proto3";

package reader;

service ReaderService {
    rpc GetTrack (GetTrackRequest) returns (GetTrackResponse);
    rpc GetTracksInAlbum (GetTracksInAlbumRequest) returns (GetTracksInAlbumResponse);
    rpc GetAlbum (GetAlbumRequest) returns (GetAlbumResponse);
    rpc GetAlbumInArtist (GetAlbumInArtistRequest) returns (GetAlbumInArtistResponse);
    rpc GetTrackGenre (GetTrackGenreRequest) returns (GetTrackGenreResponse);
}

message GetTrackRequest {
    int32 track_id = 1;
}

message GetTrackResponse {
    int32 track_id = 1;
    string title = 2;
    int32 album_id = 3;
    bool explicit = 4;
}

message GetTracksInAlbumRequest {
    int32 album_id = 1;
}

message GetTracksInAlbumResponse {
    repeated GetTrackResponse tracks = 1;
}

message GetAlbumRequest {
    int32 album_id = 1;
}

message GetAlbumResponse {
    int32 album_id = 1;
    string title = 2;
    int32 artist_id = 3;
    string release_date = 4;
}

message GetAlbumInArtistRequest {
    int32 artist_id = 1;
}

message GetAlbumInArtistResponse {
    repeated GetAlbumResponse albums = 1;
}

message GetTrackGenreRequest {
    int32 track_id = 1;
}

message GetTrackGenreResponse {
    int32 genre_id = 1;
    string genre_name = 2;
}
